src/core/components/files.js

```diff
@@ -131,22 +131,35 @@ module.exports = function files (self) {
       throw new Error('You must supply an ipfsPath')
     }
 
+    let bestMatch = { depth: -1 }
+
     ipfsPath = normalizePath(ipfsPath)
     const pathComponents = ipfsPath.split('/')
+    const prePath = normalizePath(pathComponents.slice(0, 1).join('/'))
     const restPath = normalizePath(pathComponents.slice(1).join('/'))
-    const filterFile = (file) => (restPath && file.path === restPath) || (file.path === ipfsPath)
+    const filterFile = (file) => {
+      if (file.path === ipfsPath.substring(0, file.path.length)) {
+        if (file.depth > bestMatch.depth) {
+          bestMatch = file
```

Comment:
Instead of replacing, could we retain the complete path of the best match to support more relevant error messages? (See my previous comment)

Reply:
I think replacing here is still okay as the depth tells us how many nodes `bestMatch` matches and we can use `pathComponents` to build the path. This is essentially what I was doing earlier on in the pull request. I thought of defining if a directory or file was missing, but it wasn't part of the spec so I left it the way it was.

We could have another scenario where a user tries to specify a file under a file thinking it is a directory. What do you think of these errors?

**Missing directory**
`no directory named "wrong-dir" under QmYmW4HiZhotsoSqnv2o1oUusvkRM8b9RweBoH7ao5nki2/init-docs/docs`

**Missing file**
`no file named "does-not-exist" under QmYmW4HiZhotsoSqnv2o1oUusvkRM8b9RweBoH7ao5nki2/init-docs/docs`

**File instead of directory**
`"actually-a-file" is a file not a directory under QmYmW4HiZhotsoSqnv2o1oUusvkRM8b9RweBoH7ao5nki2/init-docs/docs`

Reply:
@youngnicks this would be even better! 👍